src/server/hotkey_collector.h

```diff
@@ -79,9 +79,13 @@ class hotkey_collector : public dsn::replication::replica_base
 private:
     void on_start_detect(dsn::replication::detect_hotkey_response &resp);
     void on_stop_detect(dsn::replication::detect_hotkey_response &resp);
+    void terminate_colletor();
+    void terminate_by_timeout();
```

Comment:
How about rename to `check_timeout` ?

Reply:
It not only checked but also terminate the hotkey_collector

Reply:
how about `terminate_if_timeout ` ?